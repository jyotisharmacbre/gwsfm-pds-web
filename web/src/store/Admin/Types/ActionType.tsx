export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
