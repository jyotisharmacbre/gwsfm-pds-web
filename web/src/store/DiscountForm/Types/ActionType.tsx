export enum ActionType {
	DISCOUNT_FORM_DATA_ADD = 'DISCOUNT_FORM_DATA_ADD',
	DISCOUNT_FORM_DATA_ERROR = 'DISCOUNT_FORM_DATA_ERROR',
	DISCOUNT_FORM_DATA_EDIT = 'DISCOUNT_FORM_DATA_EDIT',
	RESET_DISCOUNT_FORM_STATE = 'RESET_DISCOUNT_FORM_STATE',
	RESET_DISCOUNT_FORM_STATE_NOTIFIER = 'RESET_DISCOUNT_FORM_STATE_NOTIFIER',
	DISCOUNT_FORM_DATA_GET = 'DISCOUNT_FORM_DATA_GET',
	DISCOUNT_FORM_DATA_GET_ERROR = 'DISCOUNT_FORM_DATA_GET_ERROR'
}
