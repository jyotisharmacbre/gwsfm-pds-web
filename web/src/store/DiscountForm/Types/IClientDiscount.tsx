export default interface IClientDiscount {
    discountId: string;
    projectId: string;
    clientState: string;
    discountType: number;
    discount: number | undefined;
    clientComments: string;
}
