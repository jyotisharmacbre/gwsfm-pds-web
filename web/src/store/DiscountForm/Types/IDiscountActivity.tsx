export interface IDiscountActivity {
  discountId: string;
  projectId: string;
  supplierName: string;
  supplierState: string;
  supplierTotalDiscount: number | undefined;
  supplierComments: string;
  clientState: string;
  discountType: number | undefined;
  clientDiscount: number  | undefined;
  clientComments: string;
}
