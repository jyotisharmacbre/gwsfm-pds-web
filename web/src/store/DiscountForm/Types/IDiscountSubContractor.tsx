export default interface IDiscountSubContractor {
    subContractorDiscountId: string;
    projectId: string;
    supplierName: string;
    supplierState: string;
    supplierTotalDiscount: number | undefined;
    supplierComments: string;
}
