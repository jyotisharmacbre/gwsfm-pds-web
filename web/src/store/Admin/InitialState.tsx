import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
