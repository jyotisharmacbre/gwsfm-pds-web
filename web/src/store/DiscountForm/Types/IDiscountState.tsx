import { IDiscountActivity } from './IDiscountActivity';
import EventType from '../../../enums/EventType';
import Notify from '../../../enums/Notify';

export interface IDiscountState {
  form: IDiscountActivity;
  error: string | null;
  loading: boolean;
  notify: Notify;
  event: EventType;
}
