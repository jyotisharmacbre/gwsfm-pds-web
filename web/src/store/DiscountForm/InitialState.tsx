import { IDiscountState } from './Types/IDiscountState';
import Notify from '../../enums/Notify';
import EventType from '../../enums/EventType';
import IDiscountSubContractor from './Types/IDiscountSubContractor';

export const newSubContractorDiscount: IDiscountSubContractor = {
  subContractorDiscountId: '',
  projectId: '',
  supplierName: '',
  supplierState: '',
  supplierTotalDiscount: undefined,
  supplierComments: '',
}

export const initialState: IDiscountState = {
  form: {
    projectId: '',
    clientDiscount: {
      discountId: '',
      projectId: '',
      discount: undefined,
      clientState: '',
      discountType: 1,
      clientComments: ''
    },
    subContractorDiscounts: [{ ...newSubContractorDiscount }],
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};