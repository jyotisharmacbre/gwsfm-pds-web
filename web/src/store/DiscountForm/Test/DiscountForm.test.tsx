import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as actions from '../../rootActions';
import { initialState } from '../InitialState';
import {ActionType} from '../../Common/Types/ActionType';
import { ActionType as discountActionType } from '../Types/ActionType'
import discountFormReducer from '../Reducer';

const mockStore = configureStore([ thunk ]);
let reduxStore;
const setUpStore = (initialState) => {
	reduxStore = mockStore({
		discount: initialState
	});
};
describe('discount for actions', () => {
	beforeEach(() => {
		setUpStore(initialState);
		reduxStore.clearActions();
	});
    it('should dispatch the default action when requested projectId data exists in store', () => {
		initialState.form.projectId = 'TEST';
		setUpStore(initialState);
		const expectedActions = [
			{
				type: ActionType.DEFAULT
			}
		];
		reduxStore.dispatch(actions.getDiscountData('TEST'));
		expect(reduxStore.getActions()).toEqual(expectedActions);
	});
});

let store = mockStore();

describe('DiscountForm reducer', () => {
	beforeEach(() => {
	  store.dispatch = jest.fn();
	})
  
	it('should handle DISCOUNT_FORM_DATA_ADD action', () => {
	  const DISCOUNT_FORM_DATA_ADD: any = {
		type: discountActionType.DISCOUNT_FORM_DATA_ADD,
		payload: {data: {id: "1"}}
	  };
	  expect(
		discountFormReducer(initialState, DISCOUNT_FORM_DATA_ADD)
	  ).toMatchSnapshot();
	});

	it('should handle DISCOUNT_FORM_DATA_EDIT action', () => {
		const DISCOUNT_FORM_DATA_EDIT: any = {
		  type: discountActionType.DISCOUNT_FORM_DATA_EDIT,
		  payload: {data: {id: "1"}}
		};
		expect(
		  discountFormReducer(initialState, DISCOUNT_FORM_DATA_EDIT)
		).toMatchSnapshot();
	  });

	  it('should handle discountFormError action', () => {
		const DISCOUNT_FORM_DATA_ERROR: any = {
		  type: discountActionType.DISCOUNT_FORM_DATA_ERROR,
		  error: "test error"
		};
		expect(
		  discountFormReducer(initialState, DISCOUNT_FORM_DATA_ERROR)
		).toMatchSnapshot();
	  });

	  it('should handle RESET_DISCOUNT_FORM_STATE action', () => {
		const RESET_DISCOUNT_FORM_STATE: any = {
		  type: discountActionType.RESET_DISCOUNT_FORM_STATE
		};
		expect(
		  discountFormReducer(initialState, RESET_DISCOUNT_FORM_STATE)
		).toMatchSnapshot();
	  });

	  it('should handle RESET_DISCOUNT_FORM_STATE_NOTIFIER action', () => {
		const RESET_DISCOUNT_FORM_STATE_NOTIFIER: any = {
		  type: discountActionType.RESET_DISCOUNT_FORM_STATE_NOTIFIER
		};
		expect(
		  discountFormReducer(initialState, RESET_DISCOUNT_FORM_STATE_NOTIFIER)
		).toMatchSnapshot();
	  });

	  it('should handle DISCOUNT_FORM_DATA_GET action', () => {
		const DISCOUNT_FORM_DATA_GET: any = {
		  type: discountActionType.DISCOUNT_FORM_DATA_GET,
		  payload: {data: {id: "1"}}
		};
		expect(
		  discountFormReducer(initialState, DISCOUNT_FORM_DATA_GET)
		).toMatchSnapshot();
	  });

	  it('should handle DISCOUNT_FORM_DATA_GET_ERROR action', () => {
		const DISCOUNT_FORM_DATA_GET_ERROR: any = {
		  type: discountActionType.DISCOUNT_FORM_DATA_GET_ERROR,
		  error: true
		};
		expect(
		  discountFormReducer(initialState, DISCOUNT_FORM_DATA_GET_ERROR)
		).toMatchSnapshot();
	  });

	  it('should handle SET_LOADING_TRUE action', () => {
		const SET_LOADING_TRUE: any = {
		  type: discountActionType.SET_LOADING_TRUE,
		  payload: {loading: true, event: 'SAVE'}
		};
		expect(
		  discountFormReducer(initialState, SET_LOADING_TRUE)
		).toMatchSnapshot();
	  });
});