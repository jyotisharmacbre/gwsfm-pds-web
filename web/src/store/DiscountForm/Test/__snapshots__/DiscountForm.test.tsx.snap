// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountForm reducer should handle DISCOUNT_FORM_DATA_ADD action 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "data": Object {
      "id": "1",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`DiscountForm reducer should handle DISCOUNT_FORM_DATA_EDIT action 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "data": Object {
      "id": "1",
    },
  },
  "loading": false,
  "notify": 1,
}
`;

exports[`DiscountForm reducer should handle DISCOUNT_FORM_DATA_GET action 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "data": Object {
      "id": "1",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": false,
  "notify": 0,
}
`;

exports[`DiscountForm reducer should handle DISCOUNT_FORM_DATA_GET_ERROR action 1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": false,
  "notify": 0,
}
`;

exports[`DiscountForm reducer should handle RESET_DISCOUNT_FORM_STATE action 1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": false,
  "notify": 0,
}
`;

exports[`DiscountForm reducer should handle RESET_DISCOUNT_FORM_STATE_NOTIFIER action 1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": false,
  "notify": 0,
}
`;

exports[`DiscountForm reducer should handle SET_LOADING_TRUE action 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": true,
  "notify": 0,
}
`;

exports[`DiscountForm reducer should handle discountFormError action 1`] = `
Object {
  "error": "test error",
  "event": 3,
  "form": Object {
    "clientDiscount": Object {
      "clientComments": "",
      "clientState": "",
      "discount": undefined,
      "discountId": "",
      "discountType": 1,
      "projectId": "",
    },
    "projectId": "TEST",
    "subContractorDiscounts": Array [
      Object {
        "projectId": "",
        "subContractorDiscountId": "",
        "supplierComments": "",
        "supplierName": "",
        "supplierState": "",
        "supplierTotalDiscount": undefined,
      },
    ],
  },
  "loading": false,
  "notify": 2,
}
`;
